docs/error-codes.md

```diff
@@ -256,4 +462,166 @@ error and warning codes.
+
+  ``` C#
+  void TestMethod(Type type)
+  {
+      // IL2054 Trim analysis: Unrecognized value passed to the parameter 'type' of method 'System.Runtime.CompilerServices.RuntimeHelpers.RunClassConstructor(RuntimeTypeHandle type)'. 
+      // It's not possible to guarantee the availability of the target static constructor.
+      RuntimeHelpers.RunClassConstructor(type.TypeHandle);
+  }
+  ```
+
+#### `IL2055` Trim analysis: Call to `System.Reflection.MethodInfo.MakeGenericMethod` can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic method.
+
+- ILLink currently doesn't analyze `MethodInfo` values and thus can't statically determine the generic method the `MakeGenericMethod` operates on. If the actual method has generic parameters with `DynamicallyAccessedMembersAttribute` ILLink would be required to fulfill the requirements declared by those attributes, but since the ILLink doesn't know the method, it can't determine if such requirements exist.  
+
+  ``` C#
+  void TestMethod()
+  {
+      // IL2055 Trim analysis: Call to `System.Reflection.MethodInfo.MakeGenericMethod` can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic method.
+      typeof(MyType).GetMethod("MyMethod").MakeGenericMethod(new Type[] { typeof(MyType) });
+  }
+  ```
+
+#### `IL2056` Trim analysis: The assembly 'assembly name' can not be found
+
+- Calling `CreateInstance` with assembly name 'assembly name' which can't be resolved.  
+
+  ``` C#
+  void TestMethod()
+  {
+      // IL2056 Trim analysis: The assembly 'NonExistentAssembly' can not be found
+      Activator.CreateInstance("NonExistentAssembly", "MyType");
+  }
+  ```
+
+#### `IL2057` Trim analysis: The type 'type name' can not be found in assembly 'assembly'
+
+- Calling `CreateInstance` with a type name which can't be resolved in the target assembly.  
+
+  ``` C#
+  void TestMethod()
+  {
+      // IL2057 Trim analysis: The type 'NonExistentType' can not be found in assembly 'MyAssembly'
+      Activator.CreateInstance("MyAssembly", "NonExistentType");
+  }
+  ```
+
+#### `IL2058` Trim analysis: Unrecognized value passed to the parameter 'typeName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+
+- The value passed as the type name to the `CreateInstance` method can't be statically analyzed, ILLink can't make sure that the type is available.  
+
+  ``` C#
+  void TestMethod(string typeName)
+  {
+      // IL2058 Trim analysis: Unrecognized value passed to the parameter 'typeName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
+      Activator.CreateInstance("MyAssembly", typeName);
+  }
+  ```
+
+#### `IL2059` Trim analysis: Unrecognized value passed to the parameter 'assemblyName' of method 'CreateInstance'. It's not possible to guarantee the availability of the target type.
```

Comment:
Just checking - we want to share 2056-2059 across various CreateInstance* methods right?

Reply:
They're already shared between AppDomain/Assembly/Activator CreateInstance methods